database/main.go

```diff
@@ -0,0 +1,66 @@
+package database
+
+import (
+	"fmt"
+	"os"
+	"sync"
+
+	_ "github.com/joho/godotenv/autoload"
+	"github.com/speedrun-website/leaderboard-backend/graph/model"
+	"gorm.io/driver/postgres"
+	"gorm.io/gorm"
+)
+
+type dbConfig struct {
+	host     string
+	port     string
+	user     string
+	dbname   string
+	password string
+}
+
+var config = dbConfig{
+	os.Getenv("POSTGRES_HOST"),
+	os.Getenv("POSTGRES_PORT"),
+	os.Getenv("POSTGRES_USER"),
+	os.Getenv("POSTGRES_DB"),
+	os.Getenv("POSTGRES_PASSWORD"),
+}
+
+var lock sync.Mutex
+var singleton *gorm.DB
+
+var dns = fmt.Sprintf(
+	"host=%s port=%s user=%s dbname=%s password=%s sslmode=disable",
+	config.host, config.port, config.user, config.dbname, config.password)
+
+func GetDatabase() (*gorm.DB, error) {
+	lock.Lock()
+	defer lock.Unlock()
+
+	if singleton != nil {
+		return singleton, nil
+	}
+
+	db, err := gorm.Open(postgres.Open(dns), &gorm.Config{})
```

Comment:
I'm not entirely sure how to handle things with Gorm. For example, [this example](https://blog.logrocket.com/how-to-build-a-rest-api-with-golang-using-gin-and-gorm/) simple exposes a `var DB *gorm.DB` in the package that is created once and then imported everywhere and that seems to work fine. 

The documentation also never talks about closing the DB, maybe we should ignore that too.

Reply:
Exporting a db variable is indeed a good way to do it. I'm not sure it matters whether the db connection is explicitly closed at shutdown because it should just happen automatically anyway.

Reply:
Alright sounds good thanks